app/response/utils.py

```diff
@@ -665,24 +661,62 @@ def _edit_email_handler(request_id, data, page, agency_name, email_template):
         # ...
     }
     editor = editor_for_type[resp.type](current_user, resp, flask_request, update=False)
+    if editor.no_change:
+        return jsonify({"error": "No changes detected."}), 200
+    else:
+        email_summary_requester, email_summary_edited, header = _get_edit_response_template(editor)
+
+    # if confirmation is not empty and response type is not FILE, store email templates into redis
+    if eval_request_bool(data.get('confirmation'), False) and resp.type != response_type.FILE:
+        email_redis.set(get_email_key(response_id), email_summary_edited)
+        if email_summary_requester is not None:
+            email_redis.set(get_email_key(response_id, requester=True), email_summary_requester)
+    return jsonify({
+        "template": email_summary_requester or email_summary_edited,
+        "header": header
+    }), 200
+
+
+def _get_edit_response_template(editor):
+    """
+    Get the email template(s) and header for confirmation page, for the edit response workflow, based on privacy options.
+
+    :param editor: editor object from class ResponseEditor
+
+    :return: email template for agency users.
+             email template for requester if privacy is not private.
+             header for confirmation page
+
+    """
     header = None
-    if confirmation:
+    data = editor.flask_request.form
+    agency_name = Requests.query.filter_by(id=editor.response.request.id).first().agency.name
+    page = urljoin(flask_request.host_url, url_for('request.view', request_id=editor.response.request.id))
+    email_template = os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], "email_edit_file.html") \
+        if editor.response.type == response_type.FILE \
+        else os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], data['template_name'])
```

Comment:
I would say you should stop passing in template_name and just infer the template from the response type